libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Memory/Expressions/DialogExpression.cs

```diff
@@ -67,5 +67,15 @@ public override void SetValue(object value)
 
             base.SetValue(value);
         }
+
+        public new string ToString()
+        {
+            if (this.Value != null)
+            {
+                return this.Value.Id;
+            }
+
+            return base.ToString();
```

Comment:
>. [](start = 23, length = 1)

This was mucking up Id calculation, was ending up with id's == type name instead of dialog id.